bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,325 @@
+** Maven plugin goals to provision server are defined in the wildfly/galleon project
+* https://github.com/wildfly/galleon-plugins[org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin]
+** Maven plugin goals to build galleon feature-packs for WildFly are defined in wildfly/galleon-plugins project
+* https://github.com/wildfly/wildfly-maven-plugin[org.wildfly.plugins:wildfly-maven-plugin]
+** Maven plugin goals to manage a WildFly server and its deployments are defined in wildfly/wildfly-maven-plugin project
+
+It makes sense to define a new maven artifact for the plugin and not extend an existing plugin:
+
+* The new plugin depends on galleon-maven-plugin for the provisioning aspects but is WildFly specific. Galleon is only one aspect of it.
+* Adding goals to wildfly-maven-plugin would create confusion (start/run/shutdown are already existing goals that would need to be named differently for the bootable jar). The deployment model is also different.
+
+So the new plugin should be a new Maven artifact with the same groupID as the wildfly-maven-plugin: org.wildfly.plugins, artifactId: wildfly-jar-maven-plugin
+A new GIT project in wildfly organization:  wildfly-jar-maven-plugin
+
+The bootable jar maven project is currently hosted in https://github.com/wildfly-extras/wildfly-jar-maven-plugin[wildfly-extras/wildfly-jar-maven-plugin]
+
+=== Hard Requirements
+
+For a complete list of plugin options, check the plugin https://github.com/wildfly-extras/wildfly-jar-maven-plugin/releases/download/2.0.0.Alpha4/index.html[community documentation].
+
+==== Build the bootable jar
+
+A new maven plugin and goal (wildfly-jar:package) is developed to output a bootable jar during maven build.
+The bootable jar is composed of:
+
+* Zipped server provisioned thanks to galleon (slim or fat server).
+* User application (war, jar, ear) if not a hollow bootable jar.
+* Runtime to launch server.
+
+The plugin output a jar named <project>-wildfly.jar
+
+===== Bootable jar runtime resolution
+
+The set of classes that implement the boot logic are retrieved at packaging time. 
+The new “org.wildfly.core:wildfly-jar-boot” jar artifact (implementation located in wildfly-core repository) 
+is resolved thanks to the wildfly-core version of the provisioned server and shaded into the bootable jar. 
+If the artifact doesn’t exist, the packaging will abort, the server being packaged not supporting bootable 
+jar packaging. (eg: old release not containing new artifacts).
+
+This artifact is in charge to unzip the server and setup a JBOSS_HOME. Once this is done, 
+it setups JBoss module classloader and call into a new JBoss module “org.wildfly.bootable-jar” 
+that is in charge of the actual server run.  This new JBoss module implementation located in wildfly-core, 
+(“org.wildfly.core:wildfly-jar-runtime” artifact), is expected to be automatically provisioned by galleon in all cases. 
+If the module is not part of the provisioned server, the packaging aborts, 
```

Comment:
IIUC it should be the server execution aborting here, rather than the `packaging`, right? Or am I missing something?

Reply:
Right, fixed.